test/jdk/sun/security/krb5/etype/WeakCrypto.java

```diff
@@ -22,61 +22,87 @@
  */
 /*
  * @test
- * @bug 6844909 8012679 8139348
+ * @bug 6844909 8012679 8139348 8273670
  * @modules java.security.jgss/sun.security.krb5
  *          java.security.jgss/sun.security.krb5.internal.crypto
+ * @library /test/lib
  * @run main/othervm WeakCrypto
- * @run main/othervm WeakCrypto true
- * @run main/othervm WeakCrypto false
- * @summary support allow_weak_crypto in krb5.conf
  */
 
-import java.lang.Exception;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.util.Arrays;
-import java.util.List;
-
+import jdk.test.lib.Asserts;
+import sun.security.krb5.Config;
 import sun.security.krb5.EncryptionKey;
 import sun.security.krb5.internal.crypto.EType;
-import sun.security.krb5.EncryptedData;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
 
 public class WeakCrypto {
+    public static void main(String[] args) throws Exception {
 
-    static List<Integer> weakOnes = List.of(
-            EncryptedData.ETYPE_DES_CBC_CRC,
-            EncryptedData.ETYPE_DES_CBC_MD5,
-            EncryptedData.ETYPE_DES3_CBC_HMAC_SHA1_KD,
-            EncryptedData.ETYPE_ARCFOUR_HMAC
-    );
+        System.setProperty("java.security.krb5.conf", "tmp.conf");
 
-    public static void main(String[] args) throws Exception {
+        test(null, null,
+                18, 17, 20, 19);    // the defaults
+        test(false, null,
+                18, 17, 20, 19);    // the defaults
+        test(true, null,
+                18, 17, 20, 19);    // the defaults
 
-        String conf = "[libdefaults]\n" +
-                (args.length > 0 ? ("allow_weak_crypto = " + args[0]) : "");
-        Files.write(Paths.get("krb5.conf"), conf.getBytes());
-        System.setProperty("java.security.krb5.conf", "krb5.conf");
+        test(null, "aes256-cts aes128-cts aes256-sha2 aes128-sha2 des3-hmac-sha1 arcfour-hmac des-cbc-crc des-cbc-md5",
```

Comment:
nit: save this into a local String variable, e.g. strongAndWeak or allETypes, and reuse this.